### PR TITLE
refactor: Migrate UserModal to React

### PR DESCRIPTION
**Pull request by thisisamir98**
Migrate UserModal to React

### CODE REVIEW

File: src/page/template/wire-main.htm [hunk 1]
```diff
@@ -34,7 +34,7 @@ <h1 class="visually-hidden" data-bind="text: t('accessibility.headings.historyIm
           searchRepository: $root.repositories.search,
           teamRepository: $root.repositories.team,
         "></group-creation-modal>
-        #include('modal/user-modal.htm')
+        <div id="user-modal"></div>
```
Comment (atomrc) [inline src/page/template/wire-main.htm]: Seems like you do not use it at all. Am I missing something?

Comment (thisisamir98) [reply src/page/template/wire-main.htm]: https://github.com/wireapp/wire-webapp/pull/13187/commits/f54d9aded32fdbb97fc063f41318dd08f33ab25d

**Review by V-Gira — APPROVED** [review]

File: src/script/page/Modals/UserModal/UserModal.tsx [hunk 2]
```diff
@@ -0,0 +1,184 @@
...
+            </h2>
+          )}
+          <Icon.Close className="modal__header__button" onClick={hide} data-uie-name="do-close" />
+        </div>
+        <div className={cx('modal__body user-modal__wrapper', {'user-modal__wrapper--max': !user && !userNotFound})}>
+          {user && (
+            <>
+              <UserDetails participant={user} isSelfVerified={isSelfVerified} classifiedDomains={classifiedDomains} />
+              <EnrichedFields user={user} showDomain={isFederated} />
+              {user.isBlockedLegalHold() && (
```
Comment (atomrc) [inline src/script/page/Modals/UserModal/UserModal.tsx]: Please use `useKoSubscirbableChildren` for all observable user properties. Else we are facing potential bugs

File: src/script/page/Modals/UserModal/UserModal.tsx [hunk 3]
```diff
@@ -0,0 +1,184 @@
...
+                <div className="modal__message" data-uie-name="status-blocked-legal-hold">
+                  {t('modalUserBlockedForLegalHold', {}, replaceLinkLegalHold)}
+                </div>
+              )}
+              {user.isBlockedLegalHold() && (
```
Comment (atomrc) [inline src/script/page/Modals/UserModal/UserModal.tsx]: `useKoSubscirbableChildren`

File: src/script/page/Modals/UserModal/UserModal.tsx [hunk 4]
```diff
@@ -0,0 +1,184 @@
...
+              </div>
+            </>
+          )}
+        </div>
+      </ModalComponent>
+    </div>
+  );
+};
+
+export default UserModalComponent;
```
Comment (atomrc) [inline src/script/page/Modals/UserModal/UserModal.tsx:161]: Maybe we do not need to export the component itself, Only the API is needed

**Review by atomrc — CHANGES_REQUESTED** [review]

File: src/script/page/Modals/UserModal/UserModal.tsx (hunk 3 above)
Comment (thisisamir98) [reply src/script/page/Modals/UserModal/UserModal.tsx]: https://github.com/wireapp/wire-webapp/pull/13187/commits/2fbd40ae0108723c10201a6a345f4c8d1a4147f2

File: src/script/page/Modals/UserModal/UserModal.tsx (hunk 2 above)
Comment (thisisamir98) [reply src/script/page/Modals/UserModal/UserModal.tsx]: https://github.com/wireapp/wire-webapp/pull/13187/commits/2fbd40ae0108723c10201a6a345f4c8d1a4147f2

File: src/script/page/Modals/UserModal/UserModal.tsx (hunk 4 above)
Comment (thisisamir98) [reply src/script/page/Modals/UserModal/UserModal.tsx:161]: needed for testing

**Review by atomrc — APPROVED** [review]

**Pull request merged**